src/NUnitFramework/framework/Internal/Builders/NUnitTestCaseBuilder.cs

```diff
@@ -203,6 +203,14 @@ private static bool CheckTestMethodSignature(TestMethod testMethod, TestCasePara
             else if (parms == null || !parms.HasExpectedResult)
                 return MarkAsNotRunnable(testMethod, "Method has non-void return value, but no result is expected");
 
+#if !THREAD_ABORT
+            if (AsyncToSyncAdapter.AcceptsCancellationToken(testMethod.Method.MethodInfo))
+            {
+                // Implict CancellationToken argument
+                argsProvided++;
```

Comment:
Will this also work if the user is already supplying the CancellationToken or will we then get a off-by-one error. E.g. if the user now have something like (probably not very likely - but could we break somebody?)

```csharp
    [TestCaseSource(nameof(TestCases))]
    public void DivideTest(int n, CancellationToken token)
    {
...
    }

    static object[] TestCases=
    {
        new object[] { 5, someCancellationToken },
        new object[] { 12, anotherCancellationToken }
    };
```

Reply:
Theoretically yes. I will update the code to check if the actual parameters already has a CancellationToken.
I'm on leave without a computer but will do this next weekend when I get back.